diff --git a/src/object/enum.cpp b/src/object/enum.cpp
--- a/src/object/enum.cpp
+++ b/src/object/enum.cpp
@@ -113,7 +113,6 @@
 #if PY_VERSION_HEX < 0x03000000
     | Py_TPFLAGS_CHECKTYPES
 #endif
-    | Py_TPFLAGS_HAVE_GC
     | Py_TPFLAGS_BASETYPE,                  /* tp_flags */
     0,                                      /* tp_doc */
     0,                                      /* tp_traverse */
