diff --git a/include/boost/math/tools/mp.hpp b/include/boost/math/tools/mp.hpp
--- a/include/boost/math/tools/mp.hpp
+++ b/include/boost/math/tools/mp.hpp
@@ -337,7 +337,7 @@
 
 // Index sequence
 // Use C++14 index sequence if available
-#if defined(__cpp_lib_integer_sequence) && (__cpp_lib_integer_sequence >= 201304)
+#if 0 && defined(__cpp_lib_integer_sequence) && (__cpp_lib_integer_sequence >= 201304)
 #include <utility>
 template<std::size_t... I>
 using index_sequence = std::index_sequence<I...>;
