Submodule libs/python contains modified content
diff --git a/libs/python/include/boost/python/operators.hpp b/libs/python/include/boost/python/operators.hpp
index ea2be7b9..9ebc3a9d 100644
--- a/libs/python/include/boost/python/operators.hpp
+++ b/libs/python/include/boost/python/operators.hpp
@@ -19,6 +19,19 @@
 # include <boost/lexical_cast.hpp>
 # include <boost/python/refcount.hpp>
 # include <boost/python/detail/unwrap_wrapper.hpp>
+// Workaround msvc iso646.h
+#if defined(_MSC_VER) && !defined(__clang__)
+#ifndef __GCCXML__
+#if defined(or)
+#   pragma push_macro("or")
+#   pragma push_macro("xor")
+#   pragma push_macro("and")
+#   undef or
+#   undef xor
+#   undef and
+#endif
+#endif
+#endif
 # include <string>
 # include <complex>
 
@@ -368,6 +381,16 @@ using boost::python::self_ns::float_;
 using boost::python::self_ns::complex_;
 using boost::python::self_ns::str;
 using boost::python::self_ns::repr;
+// Workaround msvc iso646.h
+#if defined(_MSC_VER) && !defined(__clang__)
+#ifndef __GCCXML__
+#if defined(or)
+#   pragma pop_macro("or")
+#   pragma pop_macro("and")
+#   pragma pop_macro("xor")
+#endif
+#endif
+#endif
 using boost::python::self_ns::pow;
 # endif
 
