diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0ddc96b..b978939 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -172,7 +172,7 @@ SET(${PROJECT_NAME}_SOURCES
   src/version.cpp
 )
 
-ADD_LIBRARY(${PROJECT_NAME} SHARED ${${PROJECT_NAME}_SOURCES} ${${PROJECT_NAME}_HEADERS})
+ADD_LIBRARY(${PROJECT_NAME} ${${PROJECT_NAME}_SOURCES} ${${PROJECT_NAME}_HEADERS})
 TARGET_INCLUDE_DIRECTORIES(${PROJECT_NAME}
   SYSTEM PUBLIC
   ${Boost_INCLUDE_DIRS}
diff --git a/cmake/boost.cmake b/cmake/boost.cmake
index 6177257..96988ff 100644
--- a/cmake/boost.cmake
+++ b/cmake/boost.cmake
@@ -151,7 +151,6 @@ MACRO(SEARCH_FOR_BOOST)
   FIND_PACKAGE(Boost ${BOOST_REQUIRED})
   STRING(REPLACE "_" "." Boost_SHORT_VERSION ${Boost_LIB_VERSION})
   IF("${Boost_SHORT_VERSION}" VERSION_GREATER "1.70" OR "${Boost_SHORT_VERSION}" VERSION_EQUAL "1.70")
-    SET(BUILD_SHARED_LIBS ON)
     SET(Boost_NO_BOOST_CMAKE ON)
   ENDIF("${Boost_SHORT_VERSION}" VERSION_GREATER "1.70" OR "${Boost_SHORT_VERSION}" VERSION_EQUAL "1.70")
 
diff --git a/cmake/deprecated.hh.cmake b/cmake/deprecated.hh.cmake
index 0726e3e..8fd0928 100644
--- a/cmake/deprecated.hh.cmake
+++ b/cmake/deprecated.hh.cmake
@@ -26,7 +26,7 @@
 #  define @PACKAGE_CPPNAME@_DEPRECATED __attribute__ ((deprecated))
 # else
 #  if defined(_MSC_VER) && !defined(__INTEL_COMPILER)
-#   define @PACKAGE_CPPNAME@_DEPRECATED __declspec (deprecated)
+#   define @PACKAGE_CPPNAME@_DEPRECATED /* nothing */
 #  else
 // If the compiler is not recognized, drop the feature.
 #   define @PACKAGE_CPPNAME@_DEPRECATED /* nothing */
